pkg/capabilities/triggers/mercury_trigger_test.go

```diff
@@ -246,26 +198,26 @@ func TestMultipleMercuryTriggers(t *testing.T) {
 			ObservationTimestamp: 20,
 		},
 	}
-	err = ts.ProcessReport(fr2)
+	err = ts.ProcessReport(mfr2)
 	assert.NoError(t, err)
-	assert.Len(t, callback1, 0)
-	assert.Len(t, callback2, 1)
 
-	msg = <-callback2
-	triggerEvent, reports = upwrapTriggerEvent(t, msg.Value)
-	require.NoError(t, err)
-	assert.Equal(t, "mercury", triggerEvent.TriggerType)
-	payload = make([]mercury.FeedReport, 0)
-	payload = append(payload, mfr2[0])
-	assert.Equal(t, GenerateTriggerEventID(payload), triggerEvent.ID)
-	assert.Len(t, reports, 1)
-	assert.Equal(t, mfr2[0], reports[0])
+	for rMsg := range callback2 {
+		triggerEvent, reports = upwrapTriggerEvent(t, rMsg.Value)
+		require.NoError(t, err)
+		require.Len(t, reports, 2)
+		require.Equal(t, "mercury", triggerEvent.TriggerType)
+		if reports[1].BenchmarkPrice == 50 {
+			// expect to eventually get updated feed value
```

Comment:
Added a simple max-retry limit. Every test has some timeout in CI too so maybe it's OK to just rely on that...?
A fancier way to do it is to use gomega library like core does but I'm not sure if it will work well with waiting on a channel.

Reply:
maxRetry is fine, i dont like relying on CI timeouts because they burn $$ and waste developer time